port/linux/golioth_sys_linux.c

```diff
@@ -47,6 +47,63 @@ uint64_t golioth_sys_now_ms(void)
     return now_ms - start_ms;
 }
 
+/*--------------------------------------------------
+ * Mutexes
+ *------------------------------------------------*/
+
+golioth_sys_mutex_t golioth_sys_mutex_create(void)
+{
+    pthread_mutex_t *mutex = golioth_sys_malloc(sizeof(pthread_mutex_t));
+    if (!mutex)
+    {
+        GLTH_LOGE(TAG, "Failed to allocate memory for mutex");
+        return NULL;
+    }
+
+    int err = pthread_mutex_init(mutex, NULL);
+    assert(0 == err);
```

Comment:
I don't think we should assert here. Mutex initialization is most likely to happen during system startup, so asserting here could put us into a boot loop. I think freeing and returning NULL (as below) is fine.